src/content/perf/shader.md

```diff
@@ -69,29 +40,39 @@ The following screenshot shows an example timeline tracing.
 
 ![A tracing screenshot verifying jank](/assets/images/docs/perf/render/tracing.png){:width="100%"}
 
-## What do we mean by "first run"?
+## How to reduce compilation jank
+
+If you're not using Flutter's Impeller graphics renderer and you're creating an
+iOS app, you can minimize shader compilation jank by bundling precompiled
+shaders with your iOS app. To learn more, see the iOS app development
+documentation.
 
-On iOS, "first run" means that the user might see
-jank when an animation first occurs every time
-the user opens the app from scratch.
+If you're not using Flutter's Impeller graphics renderer and you're creating an
+Android app, you can minimize shader compilation jank with the `cache-sksl`
+command line tool. To learn more, see [How to use SkSL warmup](#how-to-use-sksl-warmup).
 
 ## How to use SkSL warmup
 
-Flutter provides command line tools
-for app developers to collect shaders that might be needed
-for end-users in the SkSL (Skia Shader Language) format.
-The SkSL shaders can then be packaged into the app,
-and get warmed up (pre-compiled) when an end-user first
-opens the app, thereby reducing the compilation
-jank in later animations.
-Use the following instructions to collect
-and package the SkSL shaders:
+To reduce shader compilation jank when a user first opens an Android app, you
+can use Flutter's `cache-sksl` command-line tool. This tool precompiles shaders
+into SkSL (Skia Shading Language) format, which can then be bundled with your
+Android app.
+
+Important considerations:
+
+* On Android, precompiled shaders are device or GPU-specific.
+* The precompiled shaders for Android might only work with a subset of devices.
+* Precompiled shaders could make jank worse or produce errors on some
+  types of Android devices while making jank better on others.
+* With `cache-sksl`, you can precompile shaders for one specific class of
+  Android device at a time.
```

Comment:
If keeping this section, update "Android device" to "Android-powered device". Not super important, but the style guide does call it out: https://developers.google.com/style/word-list#android-powered